src/renderers/shared/stack/reconciler/__tests__/ReactStatelessComponent-test.js

```diff
@@ -146,6 +146,24 @@ describe('ReactStatelessComponent', function() {
     );
   });
 
+  it('should warn for functional refs in pure functions', function() {
+    spyOn(console, 'error');
+
+    var Parent = React.createClass({
+      displayName: 'Parent',
+      render: function() {
+        return <StatelessComponent name="A" ref={() => {}}/>;
+      },
+    });
+
+    ReactTestUtils.renderIntoDocument(<Parent />);
+    expect(console.error.calls.count()).toBe(1);
+    expect(console.error.calls.argsFor(0)[0]).toContain(
```

Comment:
I use `toContain` so I don't have to include the `"Warning: "` prefix in the check.